From 9b7af191ec13136985e7214b4b463641be031c20 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 7 Nov 2024 17:33:10 +0100
Subject: [PATCH 1347/1695] plat-stm32mp2: Fix RESOURCES define name

Change RESSOURCES to RESOURCES to avoid check patch issue.
with command sed -i 's/_RESSOURCES/_RESOURCES/g'

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git db1e03bcb5d4e35dcd25f2c524cf5593cd46623d plat-stm32mp2: Fix RESOURCES define name]
Fixes: d4bb9f98f4af ("plat-stm32mp2: Add PWR driver")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I843f58940ecac994eb8b7ba8249a0ec937a569e0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/417211
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../arm/plat-stm32mp2/drivers/stm32mp25_pwr.c | 46 +++++++++----------
 1 file changed, 23 insertions(+), 23 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c
index 24109924e..fd2f38808 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp25_pwr.c
@@ -79,8 +79,8 @@
 #define _PWR_SEMCR_SCID_MASK		GENMASK_32(6, 4)
 #define _PWR_SEMCR_SCID_SHIFT		U(4)
 
-#define _PWR_NB_RESSOURCES		U(13)
-#define _PWR_NB_NS_RESSOURCES		U(7)
+#define _PWR_NB_RESOURCES		U(13)
+#define _PWR_NB_NS_RESOURCES		U(7)
 #define _PWR_NB_MAX_CID_SUPPORTED	U(7)
 
 struct pwr_pdata {
@@ -110,8 +110,8 @@ static TEE_Result handle_available_semaphores(void)
 	uint32_t cidcfgr = 0;
 	unsigned int i = 0;
 
-	for (i = _PWR_NB_NS_RESSOURCES ; i < _PWR_NB_RESSOURCES; i++) {
-		unsigned int wio_offset = i + 1 - _PWR_NB_NS_RESSOURCES;
+	for (i = _PWR_NB_NS_RESOURCES ; i < _PWR_NB_RESOURCES; i++) {
+		unsigned int wio_offset = i + 1 - _PWR_NB_NS_RESOURCES;
 		vaddr_t reg_offset = pwr_d->base + _PWR_WIO_SEMCR(wio_offset);
 
 		if (!(BIT(i) & pwr_d->conf_data->access_mask[0]))
@@ -159,7 +159,7 @@ static TEE_Result apply_rif_config(bool is_tdcid)
 		return TEE_SUCCESS;
 
 	if (is_tdcid) {
-		for (i = 0; i < _PWR_NB_RESSOURCES; i++) {
+		for (i = 0; i < _PWR_NB_RESOURCES; i++) {
 			if (!(BIT(i) & pwr_d->conf_data->access_mask[0]))
 				continue;
 
@@ -169,11 +169,11 @@ static TEE_Result apply_rif_config(bool is_tdcid)
 			 * configuration prevents undesired events during the
 			 * only legitimate configuration.
 			 */
-			if (i < _PWR_NB_NS_RESSOURCES) {
+			if (i < _PWR_NB_NS_RESOURCES) {
 				io_clrbits32(pwr_d->base + _PWR_R_CIDCFGR(i),
 					     _PWR_CIDCFGR_R_CONF_MASK);
 			} else {
-				wio_offset = i + 1 - _PWR_NB_NS_RESSOURCES;
+				wio_offset = i + 1 - _PWR_NB_NS_RESOURCES;
 				io_clrbits32(pwr_d->base +
 					     _PWR_WIO_CIDCFGR(wio_offset),
 					     _PWR_CIDCFGR_W_CONF_MASK);
@@ -190,9 +190,9 @@ static TEE_Result apply_rif_config(bool is_tdcid)
 	r_sec = pwr_d->conf_data->sec_conf[0] & _PWR_R_SECCFGR_MASK;
 
 	wio_priv = (pwr_d->conf_data->priv_conf[0] &
-		    _PWR_WIO_PRIVCFGR_C_MASK) >> _PWR_NB_NS_RESSOURCES;
+		    _PWR_WIO_PRIVCFGR_C_MASK) >> _PWR_NB_NS_RESOURCES;
 	wio_sec = (pwr_d->conf_data->sec_conf[0] & _PWR_WIO_SECCFGR_C_MASK) >>
-		  _PWR_NB_NS_RESSOURCES;
+		  _PWR_NB_NS_RESOURCES;
 
 	/* Security and privilege RIF configuration */
 	io_clrsetbits32(pwr_d->base + _PWR_RPRIVCFGR, _PWR_R_PRIVCFGR_MASK,
@@ -209,16 +209,16 @@ static TEE_Result apply_rif_config(bool is_tdcid)
 		goto out;
 	}
 
-	for (i = 0; i < _PWR_NB_RESSOURCES; i++) {
+	for (i = 0; i < _PWR_NB_RESOURCES; i++) {
 		if (!(BIT(i) & pwr_d->conf_data->access_mask[0]))
 			continue;
 
-		if (i < _PWR_NB_NS_RESSOURCES) {
+		if (i < _PWR_NB_NS_RESOURCES) {
 			io_clrsetbits32(pwr_d->base + _PWR_R_CIDCFGR(i),
 					_PWR_CIDCFGR_R_CONF_MASK,
 					pwr_d->conf_data->cid_confs[i]);
 		} else {
-			wio_offset = i + 1 - _PWR_NB_NS_RESSOURCES;
+			wio_offset = i + 1 - _PWR_NB_NS_RESOURCES;
 			io_clrsetbits32(pwr_d->base +
 					_PWR_WIO_CIDCFGR(wio_offset),
 					_PWR_CIDCFGR_W_CONF_MASK,
@@ -241,7 +241,7 @@ out:
 
 		if ((io_read32(pwr_d->base + _PWR_WIOPRIVCFGR) &
 		     (pwr_d->conf_data->access_mask[0] >>
-		      _PWR_NB_NS_RESSOURCES)) != wio_priv) {
+		      _PWR_NB_NS_RESOURCES)) != wio_priv) {
 			EMSG("pwr wio resources priv conf is incorrect");
 			panic();
 		}
@@ -254,7 +254,7 @@ out:
 
 		if ((io_read32(pwr_d->base + _PWR_WIOSECCFGR) &
 		     (pwr_d->conf_data->access_mask[0] >>
-		      _PWR_NB_NS_RESSOURCES)) != wio_sec) {
+		      _PWR_NB_NS_RESOURCES)) != wio_sec) {
 			EMSG("pwr wio resources sec conf is incorrect");
 			panic();
 		}
@@ -282,13 +282,13 @@ static void parse_dt(const void *fdt, int node)
 	}
 
 	pwr_d->nb_ressources = (unsigned int)(lenp / sizeof(uint32_t));
-	assert(pwr_d->nb_ressources <= _PWR_NB_RESSOURCES);
+	assert(pwr_d->nb_ressources <= _PWR_NB_RESOURCES);
 
 	pwr_d->conf_data = calloc(1, sizeof(*pwr_d->conf_data));
 	if (!pwr_d->conf_data)
 		panic();
 
-	pwr_d->conf_data->cid_confs = calloc(_PWR_NB_RESSOURCES,
+	pwr_d->conf_data->cid_confs = calloc(_PWR_NB_RESOURCES,
 					     sizeof(uint32_t));
 	pwr_d->conf_data->sec_conf = calloc(1, sizeof(uint32_t));
 	pwr_d->conf_data->priv_conf = calloc(1, sizeof(uint32_t));
@@ -299,7 +299,7 @@ static void parse_dt(const void *fdt, int node)
 
 	for (i = 0; i < pwr_d->nb_ressources; i++)
 		stm32_rif_parse_cfg(fdt32_to_cpu(cuint[i]), pwr_d->conf_data,
-				    _PWR_NB_RESSOURCES);
+				    _PWR_NB_RESOURCES);
 
 skip_rif:
 #ifdef CFG_STM32_PWR_IRQ
@@ -325,13 +325,13 @@ static void pm_suspend(void)
 	uint32_t r_sec = 0;
 	size_t i = 0;
 
-	for (i = 0; i < _PWR_NB_RESSOURCES; i++) {
-		if (i < _PWR_NB_NS_RESSOURCES) {
+	for (i = 0; i < _PWR_NB_RESOURCES; i++) {
+		if (i < _PWR_NB_NS_RESOURCES) {
 			pwr_d->conf_data->cid_confs[i] =
 				io_read32(pwr_d->base + _PWR_R_CIDCFGR(i)) &
 				_PWR_CIDCFGR_R_CONF_MASK;
 		} else {
-			wio_offset = i + 1 - _PWR_NB_NS_RESSOURCES;
+			wio_offset = i + 1 - _PWR_NB_NS_RESOURCES;
 			pwr_d->conf_data->cid_confs[i] =
 				io_read32(pwr_d->base +
 					  _PWR_WIO_CIDCFGR(wio_offset)) &
@@ -347,15 +347,15 @@ static void pm_suspend(void)
 		  _PWR_WIO_SECCFGR_MASK;
 
 	pwr_d->conf_data->priv_conf[0] = r_priv |
-					 (wio_priv << _PWR_NB_NS_RESSOURCES);
+					 (wio_priv << _PWR_NB_NS_RESOURCES);
 	pwr_d->conf_data->sec_conf[0] = r_sec |
-					(wio_sec << _PWR_NB_NS_RESSOURCES);
+					(wio_sec << _PWR_NB_NS_RESOURCES);
 
 	/*
 	 * The access mask is modified to restore the conf for all
 	 * resources.
 	 */
-	pwr_d->conf_data->access_mask[0] = GENMASK_32(_PWR_NB_RESSOURCES - 1,
+	pwr_d->conf_data->access_mask[0] = GENMASK_32(_PWR_NB_RESOURCES - 1,
 						      0);
 }
 
-- 
2.47.2

